### PR TITLE
Fix #948 isBST in c++

### LINKED ISSUE #948: is binary search tree
An efficient and correct algorithm to check if a tree is a binary search tree or not.
I'd like to contribute to this issue in C++.

### PR DESCRIPTION
**Pull request by sakshichahal53**
Fix #948

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
Check if a binary tree is a BST or not.

#### Languages Used:
C++

#### Files Added:
isbst.cpp

### CODE REVIEW

File: isBinary_Search_Tree/isbst.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    node->data = data;
+    return (node); 
+} 
+
+bool isbst(Node* root, int min, int max)   //function to check if tree with root node is bst or not
+{ 
+    if (root == NULL)                      //empty tree is a bst
+        return true; 
+ 
+    if(root->data < min || root->data >max) return false;
```
Comment (somya-kapoor) [inline isBinary_Search_Tree/isbst.cpp]: Please fix the indentation.

Comment (somya-kapoor) [inline isBinary_Search_Tree/isbst.cpp]: Please fix the indentation.

Comment (somya-kapoor) [reply isBinary_Search_Tree/isbst.cpp]: Also, have the comments about the function, above its definition for better readability. Try following this with other general comments too.

Comment (sakshichahal53) [reply isBinary_Search_Tree/isbst.cpp]: Surely.

File: isBinary_Search_Tree/isbst.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    return (node); 
+} 
+
+//function to check if tree with root node is bst or not
+bool isbst(Node* root, int min, int max)  
+{ 
+    //empty tree is a bst
+    if (root == NULL)                      
+        return true; 
+    if(root->data < min || root->data >max)
```
Comment (marwahmanbir) [inline isBinary_Search_Tree/isbst.cpp]: Hey! Kindly fix spacing before ``max``.

File: isBinary_Search_Tree/isbst.cpp [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+//Driver Program               
+int main() 
+{                                   
+    struct Node *root = createNode(4); 
+    root->left        = createNode(2); 
+    root->right       = createNode(1); 
+    root->left->left  = createNode(5); 
+    root->left->right = createNode(6);  
+    if (isbst(root,INT8_MIN,INT8_MAX))
```
Comment (marwahmanbir) [inline isBinary_Search_Tree/isbst.cpp]: Add space after each comma.

File: isBinary_Search_Tree/isbst.cpp (hunk 2 above)
Comment (sakshichahal53) [reply isBinary_Search_Tree/isbst.cpp]: Ohkay @marwahmanbir.

**Comment by sakshichahal53** [comment]
@somya-kapoor Please review.

**Comment by somya-kapoor** [comment]
Good work @sakshichahal53 .
@jainaman224 It is approved. Please have a final look on it.

**Comment by sakshichahal53** [comment]
@jainaman224 Please review so that I could create a new PR for other opened issues.

**Pull request merged**